DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/AddDelegateToGroupController.cs

```diff
@@ -0,0 +1,151 @@
+
+    [FeatureGate(FeatureFlags.RefactoredTrackingSystem)]
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [SetDlsSubApplication(nameof(DlsSubApplication.TrackingSystem))]
+    [SetSelectedTab(nameof(NavMenuTab.Delegates))]
+    [Route("TrackingSystem/Delegates/Groups/AddDelegateToGroup")]
+    public class AddDelegateToGroupController : Controller
+    {
+        private const string AddDelegateToGroupCookieName = "AddDelegateToGroupFilter";
+        private readonly CentreCustomPromptHelper centreCustomPromptHelper;
+        private readonly IClockService clockService;
+        private readonly IGroupsDataService groupsDataService;
+        private readonly IGroupsService groupsService;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserDataService userDataService;
+
+        public AddDelegateToGroupController(
+            IJobGroupsDataService jobGroupsDataService,
+            IUserDataService userDataService,
+            CentreCustomPromptHelper centreCustomPromptHelper,
+            IGroupsService groupsService,
+            IGroupsDataService groupsDataService,
+            IClockService clockService
+        )
+        {
+            this.centreCustomPromptHelper = centreCustomPromptHelper;
+            this.jobGroupsDataService = jobGroupsDataService;
+            this.userDataService = userDataService;
+            this.groupsService = groupsService;
+            this.groupsDataService = groupsDataService;
+            this.clockService = clockService;
+        }
+
+        [Route("{groupId:int}")]
+        public IActionResult Index(
+            int groupId = 1,
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = BaseSearchablePageViewModel.Ascending,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1
+        )
+        {
+            sortBy ??= DefaultSortByOptions.Name.PropertyName;
+            filterBy = FilteringHelper.GetFilterBy(
+                filterBy,
+                filterValue,
+                Request,
+                AddDelegateToGroupCookieName,
+                DelegateActiveStatusFilterOptions.IsActive.FilterValue
+            );
+
+            var centreId = User.GetCentreId();
+            var jobGroups = jobGroupsDataService.GetJobGroupsAlphabetical();
+            var customPrompts = centreCustomPromptHelper.GetCustomPromptsForCentre(centreId);
+            var delegateUsers = userDataService.GetDelegatesNotRegisteredForGroupByGroupId(groupId, centreId);
+            var groupName = userDataService.GetGroupNameById(groupId);
+
+            var model = new AddDelegateToGroupViewModel(
+                delegateUsers,
+                jobGroups,
+                customPrompts,
+                page,
+                groupId,
+                groupName!,
+                searchString,
+                sortBy,

```

Comment:
`sortBy` and `sortDirection` not required here as internally we should be defaulting to one.

Reply:
This has now been refactored as per the original comment.